crates/core/src/client/config.rs

```diff
@@ -36,6 +37,7 @@ impl Network {
         match self {
             Network::Katana => Ok(Url::parse(KATANA_RPC_URL)?),
             Network::Madara => Ok(Url::parse(MADARA_RPC_URL)?),
+            Network::Sharingan => Ok(Url::parse(std::env::var("SHARINGAN_RPC_URL").unwrap().as_str())?),
```

Comment:
This means `SHARINGAN_RPC_URL` needs to be passed when network used is sharingan